homeassistant/components/bluetooth/scanner.py

```diff
@@ -231,17 +238,35 @@ async def _async_start(self) -> None:
                     f"{self.name}: Timed out starting Bluetooth after {START_TIMEOUT} seconds"
                 ) from ex
             except BleakError as ex:
+                error_str = str(ex)
                 if attempt == 0:
-                    error_str = str(ex)
                     if any(
                         needs_reset_error in error_str
                         for needs_reset_error in NEED_RESET_ERRORS
                     ):
                         await self._async_reset_adapter()
                     continue
+                if attempt != START_ATTEMPTS - 1:
+                    # If we are not out of retry attempts, and the
+                    # adapter is still initializing, wait a bit and try again.
+                    if any(
+                        wait_error in error_str
+                        for wait_error in WAIT_FOR_ADAPTER_TO_INIT_ERRORS
+                    ):
+                        _LOGGER.debug(
+                            "%s: Waiting for adapter to initialize; attempt (%s/%s)",
+                            self.name,
+                            attempt + 1,
+                            START_ATTEMPTS,
+                        )
+                        await asyncio.sleep(ADAPTER_INIT_TIME)
+                        continue
+
                 _LOGGER.debug(
-                    "%s: BleakError while starting bluetooth: %s",
+                    "%s: BleakError while starting bluetooth; attempt: (%s/%s): %s",
                     self.name,
+                    attempt + 1,
```

Comment:
isn't this a bit confusing when you're reading the log later and debugging ?

Reply:
Since its 0 indexed and humans think in 1 indexed I wanted to say attempt (1/3),(2/3),(3/3) instead of (0/3) .etc

Reply:
Personal opinion I guess but changing the data in the log itself has bit me once so my personal approach would be to have a startindex of 1 in such situations and log the actual value.